apps/src/javalab/JavabuilderConnection.js

```diff
@@ -45,27 +47,41 @@ export default class JavabuilderConnection {
     // that has not been modified from the starter code.
     // This case does not apply to students, who are able to execute unmodified starter code.
     // See this comment for more detail: https://github.com/code-dot-org/code-dot-org/pull/42313#discussion_r701417221
-    if (project.getCurrentId() === undefined) {
+    if (project.getCurrentId() === undefined && !this.overrideSources) {
       this.onOutputMessage(javalabMsg.errorProjectNotEditedYet());
       return;
     }
 
     this.onOutputMessage(`${STATUS_MESSAGE_PREFIX} ${javalabMsg.connecting()}`);
     this.onNewlineMessage();
 
+    const payload = {
+      projectUrl: this.overrideSources
+        ? 'override'
+        : project.getProjectSourcesUrl(),
+      projectVersion: this.overrideSources
+        ? 'override'
+        : project.getCurrentSourceVersionId(),
```

Comment:
I was about to ask about the `override` string here, but I think that's what you're referring to in your description. i.e. that you're removing these fields alltogether, right?

Reply:
correct! PR incoming :)